clinica/pipelines/statistics_surface/_model.py

```diff
@@ -0,0 +1,728 @@
+
+DEFAULT_THRESHOLD_UNCORRECTED_P_VALUE = 0.001
+DEFAULT_THRESHOLD_CORRECTED_P_VALUE = 0.05
+DEFAULT_CLUSTER_THRESHOLD = 0.001
+MISSING_TERM_ERROR_MSG = Template(
+    "Term ${term} from the design matrix is not in the columns of the "
+    "provided TSV file. Please make sure that there is no typo."
+)
+
+
+def _print_clusters(model: SLM, threshold: float) -> None:
+    """This function prints the results related to total number
+    of clusters, as well as the significative clusters.
+
+    Parameters
+    ----------
+    model : Fitted SLM model.
+    threshold : Cluster defining threshold.
+    """
+    cprint("#" * 40)
+    cprint("After correction (Cluster-wise Correction for Multiple Comparisons): ")
+    df = model.P["clus"][0]
+    cprint(df)
+    cprint(f"Clusters found: {len(df)}")
+    cprint(
+        f"Significative clusters (after correction): {len(df[df['P'] <= threshold])}"
+    )
+
+
+def _is_categorical(df: pd.DataFrame, column: str) -> bool:
+    """This function returns whether the provided DataFrame's column
+    is categorical or not.
+
+    .. note::
+        There might be more clever ways to do that.
+
+    Parameters
+    ----------
+    df : DataFrame to analyze.
+    column : Name of the column to check.
+
+    Returns
+    -------
+    True if the column contains categorical values, False otherwise.
+    """
+    if column not in df.columns:
+        raise ValueError(MISSING_TERM_ERROR_MSG.safe_substitute(term=column))
+    return not df[column].dtype.name.startswith("float")
+
+
+def _build_model(design_matrix: str, df: pd.DataFrame) -> FixedEffect:
+    """Build a brainstat model from the design matrix in
+    string format.
+
+    This function assumes that the design matrix is formatted
+    in the following way:
+
+        1 + factor_1 + factor_2 + ...
+
+    Or:
+
+        factor_1 + factor_2 + ...
+
+    in the latter case the intercept will be added automatically.
+
+    Parameters
+    ----------
+    design_matrix : Design matrix specified as a string.
+    df : Subjects DataFrame.
+
+    Returns
+    -------
+    model : BrainStats model.
+    """
+    if len(design_matrix) == 0:
+        raise ValueError("Design matrix cannot be empty.")
+    if "+" in design_matrix:
+        terms = [_.strip() for _ in design_matrix.split("+")]
+    else:
+        terms = [design_matrix.strip()]
+    model = []
+    for term in terms:
+        # Intercept is automatically included in brainstat
+        if term == "1":
+            continue
+        # Handles the interaction effects
+        if "*" in term:
+            sub_terms = [_.strip() for _ in term.split("*")]
+            model_term = reduce(
+                lambda x, y: x * y, [_build_model_term(_, df) for _ in sub_terms]
+            )
+        else:
+            model_term = _build_model_term(term, df)
+        model.append(model_term)
+    if len(model) == 1:
+        return model[0]
+    return reduce(lambda x, y: x + y, model)
```

Comment:
Since `FixedEffect` implements `__add__`, we can fold with `sum`:

```suggestion
    return sum(model)
```